File: find_first_word.go
package main

import (
	"fmt"
)

func findFirstWord(s string) (first string) {
	for _, char := range s {
		if string(char) == " " {
			break
		}
		first += string(char)
	}
	return
}

func main() {
	fmt.Println(findFirstWord("My name is"))
}

File: RustProject/hello-rust/src/demo/find_first_word.rs
sfn find_first_word(s: &String) -> String {
	let mut ss: String = String::from("");
	for c in s.chars() {
		if c == ' ' {
			break;		
		}
		ss.push(c);
	}
	return ss;
}

fn main() {
	let str = String::from("my name");
	let ret = find_first_word(&str);
	println!("{}", ret);
}
